desktop/apps/auction2/routes.js

```diff
@@ -1,76 +1,133 @@
+import * as actions from 'desktop/apps/auction2/actions'
 import Articles from 'desktop/collections/articles.coffee'
-import ArticlesQuery from './queries/articles'
+import ArticlesQuery from 'desktop/apps/auction2/utils/queries/articles'
 import Auction from 'desktop/models/auction.coffee'
-import MeQuery from './queries/me'
-import SaleQuery from './queries/sale'
-import footerItems from './footer_items'
+import App from 'desktop/apps/auction2/components/App'
+import MeQuery from 'desktop/apps/auction2/utils/queries/me'
+import React from 'react'
+import SaleQuery from 'desktop/apps/auction2/utils/queries/sale'
+import auctionReducer, { initialState } from 'desktop/apps/auction2/reducers'
+import configureStore from 'desktop/apps/auction2/utils/configureStore'
+import footerItems from 'desktop/apps/auction2/utils/footerItems'
 import metaphysics from 'lib/metaphysics.coffee'
 import get from 'lodash.get'
-import renderTemplate from 'desktop/components/react/utils/render_template'
+import renderReactLayout from 'desktop/components/react/utils/render_react_layout'
+import u from 'updeep'
 import { getLiveAuctionUrl } from 'utils/domain/auctions/urls'
 
 export async function index (req, res, next) {
-  const { me } = await metaphysics({ query: MeQuery(req.params.id), req: req })
-  const { sale } = await metaphysics({ query: SaleQuery(req.params.id) })
-  const { articles } = await metaphysics({ query: ArticlesQuery(sale._id) })
+  const saleId = req.params.id
 
-  res.locals.sd.AUCTION = sale
+  try {
+    const {
+      sale
+    } = await metaphysics({
+      query: SaleQuery(saleId)
+    })
+
+    res.locals.sd.AUCTION = sale
+    fetchUser(req, res)
+
+    const store = configureStore(auctionReducer, {
+      auctionArtworks: u({
+        filterParams: {
+          sale_id: saleId
+        },
+        isClosed: sale.is_closed,
+        user: res.locals.sd.CURRENT_USER
+      }, initialState)
+    })
+
+    const [
+      { me },
+      { articles }
+    ] = await Promise.all([
+      metaphysics({ query: MeQuery(sale.id), req: req }),
+      metaphysics({ query: ArticlesQuery(sale._id) }),
+
+      // Hydrate store
+      store.dispatch(actions.fetchArtworksByFollowedArtists()),
+      store.dispatch(actions.fetchArtworks())
```

Comment:
It's much nicer coming to the page and having everything rendered and in place rather than having to see a spinner (point of all this work). Also, a) things are now being fetched in parallel so it actually feels faster; and b) to optimize we should cut down on the number of initial artworks requested; there's like 50 or so.